src/Messenger/IndexationRequestSpoolSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace JoliCode\Elastically\Messenger;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * Same idea as https://github.com/symfony/swiftmailer-bundle/blob/master/EventListener/EmailSenderListener.php.
+ */
+class IndexationRequestSpoolSubscriber implements EventSubscriberInterface
+{
+    private $logger;
+
+    private $wasExceptionThrown = false;
+
+    private $singleTransport;
+
+    private $bus;
+
+    public function __construct(TransportInterface $singleTransport, MessageBusInterface $bus, LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+        $this->singleTransport = $singleTransport;
+        $this->bus = $bus;
+    }
+
+    public function onException()
+    {
+        $this->wasExceptionThrown = true;
+    }
+
+    public function onTerminate()
+    {
+        if ($this->wasExceptionThrown) {
+            return;
+        }
+
+        $operations = [];
+
+        foreach ($this->singleTransport->get() as $envelope) {
+            $operations[] = $envelope->getMessage();
+
+            $this->singleTransport->ack($envelope);
+        }
+
+        if (empty($operations)) {
+            return;
+        }
+
+        $message = new MultipleIndexationRequest($operations);
+        $this->bus->dispatch($message);
+    }
+
+    public static function getSubscribedEvents()
+    {
+        $listeners = [
+            KernelEvents::EXCEPTION => 'onException',
+            KernelEvents::TERMINATE => 'onTerminate',
```

Comment:
I would listen kernel.response instead of this event

Reply:
It's better for debug yes, 
but it's not really intended for that IMO.

> This event allows you to modify or replace the response that will be replied.

Any other opinions about that?

Reply:
We don't really care :) it's just a generic description.
There is an event, we can leverage it 
If you prefer, you can use `kernel.finish_request`, but it's almost the same things...